graphql_compiler/query_pagination/query_parameterizer.py

```diff
@@ -72,21 +84,18 @@ def _is_new_filter_stronger(operation: str, new_filter_value: Any, old_filter_va
     Returns:
         whether the old filter can be removed with no change in query meaning.
     """
-    if type(new_filter_value) != type(old_filter_value):
-        raise AssertionError(
-            f"Expected {new_filter_value} and {old_filter_value} "
-            f"to have the same type, but got {type(new_filter_value)} "
-            f"and {type(old_filter_value)}."
-        )
+    vertex_type = query_analysis.types[property_path.vertex_path]
```

Comment:
Isn't this just `query_analysis.metadata_table.get_location_info(...).type.name` or something like that?

The new property and function to compute it don't seem particularly useful to me, I feel like I'm missing something.

Reply:
The metadata table uses `BaseLocation` for the key. For this purpose it's fine to make up `visit_counter = 1`, but in general I don't want to deal with gremlin artifacts in this module. In fact, I suspect there's a bug somewhere because of the way we handle `visit_counter`, and I'm looking for it.

Reply:
See https://github.com/kensho-technologies/graphql-compiler/pull/775